src/contracts/strategies/StrategyBaseTVLLimits.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "./StrategyBase.sol";
+
+/**
+ * @title A Strategy implementation inheriting from `StrategyBase` that limits the total amount of deposits it will accept.
+ * @author Layr Labs, Inc.
+ */
+contract StrategyBaseTVLLimits is StrategyBase {
+    /// The maximum deposit (in underlyingToken) that this strategy will accept per deposit
+    uint256 public maxPerDeposit;
```

Comment:
seems a bit like overkill to also have this, but we can discuss

Reply:
A max per deposit makes it so that it is annoying for a whale to stake entirely to the maxDeposits for the strategy. This allows small stakers to get in by making it anoying for whales and eating into their costst with gas.

Reply:
Agreed offline it's OK to add